### PR TITLE
Return meaningful error from get unsubsidized fee

### PR DESCRIPTION
**Pull request by vkgnosis**
Previously we were treating all errors as InsufficientFee which hides
real back end errors showing up.

Note that consuming code in order_validation.rs still ignores the error.

### Test Plan

CI

### CODE REVIEW

File: crates/orderbook/src/fee.rs [hunk 1]
```diff
@@ -342,15 +355,13 @@ impl MinFeeCalculating for MinFeeCalculator {
...
+    ) -> Result<FeeParameters, GetUnsubsidizedMinFeeError> {
         let cow_factor = self.cow_subsidy.cow_subsidy_factor(user);
         let past_fee = self
             .measurements
             .find_measurement_including_larger_amount(fee_data, (self.now)());
         // TODO: make this function return a proper error distinguishing errors from fee too low;
-        let (cow_factor, past_fee) = futures::future::try_join(cow_factor, past_fee)
-            .await
-            .map_err(|_| ())?;
+        let (cow_factor, past_fee) = futures::future::try_join(cow_factor, past_fee).await?;
```
Comment (MartinquaXD) [inline crates/orderbook/src/fee.rs:364]: Is that `TODO` still relevant? It looks like the error propagation is now correct, no?

**Review by MartinquaXD — APPROVED** [review]
LGTM

**Review by sunce86 — APPROVED** [review]
lg

**Review by sunce86 — APPROVED** [review]
lg

**Pull request merged**